src/compiler/transformers/classFields.ts

```diff
@@ -292,7 +536,9 @@ namespace ts {
         }
 
         function transformConstructorBody(node: ClassDeclaration | ClassExpression, constructor: ConstructorDeclaration | undefined, isDerivedClass: boolean) {
-            const properties = getInitializedProperties(node, /*isStatic*/ false);
+            const properties = node.members.filter(
+                (node): node is PropertyDeclaration => isPropertyDeclaration(node) && !hasStaticModifier(node)

```

Comment:
I would extract this to a function declaration outside of `transformConstructorBody` to avoid the possible overhead of allocating a new function object each time `transformConstructorBody` is called.